tests/Unit/Helpers/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/PrimReconstructor.hpp

```diff
@@ -0,0 +1,458 @@
+        }
+      }
+    }
+    auto& phi = get<GeneralizedHarmonic::Tags::Phi<3>>(vars);
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t a = 0; a < 4; ++a) {
+        for (size_t b = a; b < 4; ++b) {
+          phi.get(i, a, b) = (10 * i + 50 * a + 1000 * b + 1) * coords.get(i);
+        }
+      }
+    }
+
+    auto& pi = get<GeneralizedHarmonic::Tags::Pi<3>>(vars);
+    for (size_t a = 0; a < 4; ++a) {
+      for (size_t b = a; b < 4; ++b) {
+        pi.get(a, b) = (500 * a + 10000 * b + 1) * get<0>(coords);
+        for (size_t i = 1; i < 3; ++i) {
+          pi.get(a, b) += (500 * a + 10000 * b + 1 + i) * coords.get(i);
+        }
+      }
+    }
+    return vars;
+  };
+
+  const FixedHashMap<maximum_number_of_neighbors(3) + 1,
+                     std::pair<Direction<3>, ElementId<3>>,
+                     evolution::dg::subcell::NeighborData,
+                     boost::hash<std::pair<Direction<3>, ElementId<3>>>>
+      neighbor_data = compute_neighbor_data(
+          subcell_mesh, logical_coords, element.neighbors(),
+          reconstructor.ghost_zone_size(), compute_solution);
+
+  const size_t reconstructed_num_pts =
+      (subcell_mesh.extents(0) + 1) *
+      subcell_mesh.extents().slice_away(0).product();
+
+  using dg_package_data_argument_tags = tmpl::remove_duplicates<tmpl::append<
+      cons_tags,
+      tmpl::push_back<
+          prims_tags,
+          hydro::Tags::LorentzFactorTimesSpatialVelocity<DataVector, 3>>,
+      flux_tags,
+      tmpl::push_back<
+          tmpl::list<Lapse, Shift, SpatialMetric, SqrtDetSpatialMetric,
+                     InverseSpatialMetric,
+                     evolution::dg::Actions::detail::NormalVector<3>>>>>;
+
+  std::array<Variables<dg_package_data_argument_tags>, 3> vars_on_lower_face =
+      make_array<3>(
+          Variables<dg_package_data_argument_tags>(reconstructed_num_pts));
+  std::array<Variables<dg_package_data_argument_tags>, 3> vars_on_upper_face =
+      make_array<3>(
+          Variables<dg_package_data_argument_tags>(reconstructed_num_pts));
+
+  Variables<prims_tags> volume_prims{subcell_mesh.number_of_grid_points()};
+  Variables<cons_tags> volume_cons_vars{subcell_mesh.number_of_grid_points()};
+  Variables<spacetime_tags> volume_spacetime_vars{
+      subcell_mesh.number_of_grid_points()};
+  {
+    const auto volume_prims_for_recons = compute_solution(logical_coords);
+    tmpl::for_each<tmpl::list<Rho, Pressure, MagField, Phi>>(
+        [&volume_prims, &volume_prims_for_recons](auto tag_v) {
+          using tag = tmpl::type_from<decltype(tag_v)>;
+          get<tag>(volume_prims) = get<tag>(volume_prims_for_recons);
+        });
+    get<gr::Tags::SpacetimeMetric<3>>(volume_cons_vars) =
+        get<gr::Tags::SpacetimeMetric<3>>(volume_prims_for_recons);
+    const auto spatial_metric =
+        gr::spatial_metric(get<gr::Tags::SpacetimeMetric<3>>(volume_cons_vars));
+
+    // Metric is identity at cell center
```

Comment:
I didn't understand this statement. Wasn't spacetime metric set to be non-Euclidean at the line 175?